### ISSUE TITLE
Keycloak status is not a subresource

### ISSUE THREAD

**Issue by shawkins** [keycloak/keycloak#21063]
### Description

The crd for keycloak and realmimport contain the status definition as a sibling to spec, and not in the subresource status definition.  This means that on regular mutative operations the status will be updated as well, which is unexpected.

**Comment by shawkins** [comment]
@andreaTP is this a known issue with the crd generator?

**Comment by andreaTP** [comment]
> is this a known issue with the crd generator?

not that I'm aware of.

> This means that on regular mutative operations the status will be updated as well

This is not exact I think... the `status` stanza has specific(and separate) endpoints to update it: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource

From what I can see, the `status` is listed in the `subresource` and should be enough.

**Comment by shawkins** [comment]
> From what I can see, the status is listed in the subresource and should be enough.

Indeed it is.  What happened was I had left the status defined on a resource while testing the migration effects of changing the status boolean to a string.  I attempted a server side apply which validated the status as well - I didn't expect that given that would not be updated.  In checking the crd I saw that scale was defined in the subresources, but status was not and assumed that was the issue.  A couple more checks of the crd examples would have cleared up that yes this is the expected structure and that yes even if it won't be updated if the resource contains a status it will still be validated - more that likely this applies the other direction as well when using the status endpoints.
